classes/event_processor.php

```diff
@@ -755,12 +755,33 @@ public static function rerun_all_error_runs($workflow, $historic = false) {
         $ids = $DB->get_records_sql($sql, [$timelimit, $workflow]);
 
         // We now need to iterate through, and rerun.
+        $results = [];
         foreach ($ids as $run) {
             if ($historic) {
                 self::execute_workflow_from_event_historic($run->id);
             } else {
                 self::execute_workflow_from_event_current($run->id);
             }
+
+            // Get new run ID, and store the result.
+            $newrun = $DB->get_records('tool_trigger_workflow_hist', ['eventid' => $run->eventid], 'id DESC', '*', 0, 1);
```

Comment:
I wonder if it would be cleaner if the two sibling methods execute_workflow_from_event_current and execute_workflow_from_event_historic returned the new run id?

Reply:
That was my initial thought, but as they use the full event processor workflow, there is no way to do it cleanly